src/iccpd/src/mlacp_sync_update.c

```diff
@@ -181,164 +246,282 @@ int mlacp_fsm_update_mac_entry_from_peer( struct CSM* csm, struct mLACPMACData *
+                                " interface  %s, MAC %s vlan-id %d ", mac_msg->ifname,
+                                mac_addr_to_str(mac_msg->mac_addr), mac_msg->vid);
+                            return 0;
+                        }
 
-                        if (strlen(csm->peer_itf_name) != 0)
+                        if (csm->peer_link_if && (csm->peer_link_if->state == PORT_STATE_UP))
                         {
-                            if (strcmp(mac_msg->ifname, csm->peer_itf_name) == 0)
-                            {
-                                /* This MAC is already point to peer-link */
-                                return 0;
-                            }
+                            /*Redirect the mac to peer-link*/
+                            memcpy(&mac_msg->ifname, csm->peer_itf_name, MAX_L_PORT_NAME);
 
-                            if (csm->peer_link_if && csm->peer_link_if->state == PORT_STATE_UP)
-                            {
-                                /*Redirect the mac to peer-link*/
-                                memcpy(&mac_msg->ifname, csm->peer_itf_name, IFNAMSIZ);
+                            /*Send mac add message to mclagsyncd*/
+                            add_mac_to_chip(mac_msg, mac_msg->fdb_type);
 
-                                /*Send mac add message to mclagsyncd*/
-                                add_mac_to_chip(mac_msg, MAC_TYPE_DYNAMIC);
-                            }
-                            else
-                            {
-                                /*must redirect but peerlink is down, del mac from ASIC*/
-                                /*if peerlink change to up, mac will add back to ASIC*/
-                                del_mac_from_chip(mac_msg);
+                            ICCPD_LOG_DEBUG("ICCP_FDB", "Remote MAC ADD , local mac exist move to peer link up "
+                                " interface  %s, MAC %s vlan-id %d ", mac_msg->ifname,
+                                mac_addr_to_str(mac_msg->mac_addr), mac_msg->vid);
 
-                                /*Redirect the mac to peer-link*/
-                                memcpy(&mac_msg->ifname, csm->peer_itf_name, IFNAMSIZ);
-                            }
                         }
                         else
                         {
-                            /*must redirect but no peerlink, del mac from ASIC*/
-                            del_mac_from_chip(mac_msg);
+                            /*Redirect the mac to peer-link, if peerlink is down FdbOrch deletes MAC*/
+                            memcpy(&mac_msg->ifname, csm->peer_itf_name, MAX_L_PORT_NAME);
 
-                            /*Update local item*/
-                            memcpy(&mac_msg->ifname, MacData->ifname, MAX_L_PORT_NAME);
+                            add_mac_to_chip(mac_msg, mac_msg->fdb_type);
```

Comment:
@Praveen-Brcm 
You changed the behavior from del_mac_from_chip() to add_mac_to_chip() even when the peerlink is down. Is it intended? Why?
Because the behavior is different from the behavior at line#531, the mac address is added when the peerlink is up.
Thank you!